### ISSUE TITLE
skip-invalid-size-inputs-valid-test for test and valid 

### ISSUE THREAD

**Issue by aastha19** [pytorch/fairseq#1202]
As described in the [docs](https://fairseq.readthedocs.io/en/latest/command_line_tools.html#Dataset%20and%20data%20loading) you can add the flag `--skip-invalid-sizes`

_Originally posted by @lematt1991 in https://github.com/pytorch/fairseq/issues/1191#issuecomment-535940389_

Using --skip-invalid-size-inputs-valid-test does not work while training, it is supposed to work for valid and test set. 
Could you please suggest any way to work around the Assertion error.

Also, could you please let me know a better approach to train 25 million training dataset as it is getting impossible to train such a huge dataset in one go.

**Comment by myleott** [comment]
I thought we automatically skipped inputs that were too long in the training set (this wasn’t true for valid/test, thus the need for the skip-...-valid-test option). Did something break this behavior for training data? Maybe one of the recent cython changes? cc @ngoyal2707

**Comment by aastha19** [comment]
@myleott @ngoyal2707 any update?

**Comment by aastha19** [comment]
![image](https://user-images.githubusercontent.com/14852345/65949887-a355a780-e45a-11e9-8d81-f7bcee3a58fe.png)
What does this imply?

**Comment by ngoyal2707** [comment]
`--skip-invalid-size-inputs-valid-test` only applies to `valid` and `test` phase as the name suggests.
For train phase we always skip invalid input by default.

Your logs indicate that the training successfully started but you are getting OOM on GPU.
Can you try defining `--max-tokens`, `--max-sentences`, probably to lesser value than what you currently have.

`--max-sentences` refers to maximum number of sequences that a batch can have (similar to batch size), `--max-tokens` is the maximum tokens a batch can have.
